src/generator.rs

```diff
@@ -401,7 +416,42 @@ mod tests {
             vec![],
         );
         let vec_u32_type_id = builder.register_type(vec_u32_type);
-        assert_eq!(VEC_U32_TY_ID, vec_u32_type_id);
+        let registry = builder.finish();
+
+        // build a generator for that registry
+        let generator = Generator::new(&registry);
+        // A constant fuzzer containing only 0x41 as u8 constants
+        let mut fuzzer = Fuzzer::new(
+            0,
+            Constants {
+                u32_constants: vec![0x41],
+                ..Default::default()
+            },
+        );
+
+        // get the same type we just register from the registry
+        let ty = registry.resolve(vec_u32_type_id).unwrap().type_def.clone();
+
+        // Generate a fuzzed encoded data for it
+        let encoded = generator.generate(&mut fuzzer, &vec![ty]).unwrap();
+        let parsing_pointer = &mut &encoded[..];
+
+        // Decode it by hand
+        let len = <Compact<u32>>::decode(parsing_pointer).unwrap().into();
+        assert!(len > 0); // At least one element with this seed
+        for _ in 0..len {
+            // All elements are taken from the constants (only 0x41)
+            assert_eq!(u32::decode(parsing_pointer).unwrap(), 0x41);
```

Comment:
parsing_pointer is never incremented. Isn't it  always pointing to the same memory location?